code/mob/living/silicon/ai.dm

```diff
@@ -2580,6 +2595,31 @@ proc/get_mobs_trackable_by_AI()
 	src.internal_pda.name = "[src.name]'s Internal PDA Unit"
 	src.internal_pda.owner = "[src.name]"
 
+// For if an AI needs to disconnect, make their core a latejoin one
+/mob/living/silicon/ai/proc/become_latejoin(var/announce = FALSE)
+	if (deployed_to_eyecam)
+		eyecam.return_mainframe()
+	if (deployed_shell)
+		src.return_to(deployed_shell)
+	src.ghostize()
+	if (src.mind)
+		src.mind.register_death()
+
+	//Tell the crew the AI is gone
+	if(announce)
+		command_alert("Station AI unit malfunction detected, attempting download of new mind from Central Command database.","Artificial Intelligence Update", alert_origin = ALERT_STATION)
+	logTheThing(LOG_COMBAT, src, "is replaced with a latejoin AI at [log_loc(src)].")
+
+	qdel(src.brain)
+	src.brain = new /obj/item/organ/brain/latejoin(src)
+	src.set_color(000000)
+	src.faceEmotion = "ai_blank"
+	src.update_appearance()
+	src.name = "AI"
+	src.UpdateName()
+	if (src.mind)
+		src.mind.assigned_role = "AI"
```

Comment:
Not sure what this is doing, but again ghostize should have removed the mind by this point